remove "authentication_uri" from things configured by session

rjbs opened the pull request:
This isn't stored in the session, so we'll always end up clearing it, which is no good!

rjbs committed:
remove "authentication_uri" from things configured by session

...it is not in there!

wolfsage commented:
Topicbox and helpdesk at least depend on this being here, so I think we need a different sort of fix

rjbs committed:
URL signing: sign the digest of the URL

...not the URL

rjbs committed:
URL signing: clip signed time to latest hour

... to be nice to caching

rjbs commented:
@wolfsage I think the way forward is to fix the other test suites to work with this clearly(?) correct change.  I filed a Topicbox MR with a single line fix.  (I'm re-running the test suite, but it appears correct.)

mmcclimon reviewed (approved):
I think this is fine once we fix all the downstream code. (I'll assemble a list on not-GitHub.)

rjbs committed:
URL signing: punt on exactly what to sign

some systems (notably Fastmail and Topicbox) have slightly different
signature targets

rjbs committed:
Merge branch 'fix-url-signing'
